satpy/readers/mirs.py

```diff
@@ -173,10 +173,10 @@ def limb_correct_atms_bt(bt_data, surf_type_mask, coeff_fns, ds_info):
     is_sea = (surf_type_mask == 0)
     new_data = da.where(is_sea, sea_bt, land_bt)
 
-    bt_corrected = xr.DataArray(new_data, dims=bt_data[idx, :, :].dims,
-                                coords=bt_data[idx, :, :].coords,
+    bt_corrected = xr.DataArray(new_data, dims=("y", "x"),
+                                coords=surf_type_mask.coords,
```

Comment:
What are the coords this variable has?

Reply:
I was thinking about this and am not sure the best way to add the coords.  I was considering adjusting the way the method new_coords works at line 211 to make this a more obvious way of assigning longitude and latitude coordinates.

Reply:
Actually was thinking about that same thing for the chunk size for the  map_blocks.  If I added a general self.nc.shape so that I can access the scanlines/fov without slicing the bt_data.

Reply:
Let's talk about it at the meeting today *BUT* you shouldn't need to add lon/lat/x/y coordinates. The base reader class will do that for you based on the `coordinates:` defined in the YAML (or your available_datasets method).